packages/poaps/src/PoapsClient.ts

```diff
@@ -90,4 +108,145 @@ export class PoapsClient {
       nextCursor(poaps.length, limit, offset),
     );
   }
+
+  /**
+   * Retrieves the secret code associated with a POAP code.
+   * @async
+   * @param {string} mintCode - The POAP code for which to get the secret.
+   * @returns {Promise<string>} The associated secret code.
+   * @throws {CodeAlreadyMintedError} Thrown when the POAP code has already been minted.
+   * @throws {CodeExpiredError} Thrown when the POAP code is expired.
+   */
+  private async getSecretCode(mintCode: string): Promise<string> {
+    const getCodeResponse = await this.getMintCode(mintCode);
+
+    if (getCodeResponse.claimed == true) {
+      throw new CodeAlreadyMintedError(mintCode);
```

Comment:
Then should this be minted instead of claimed? Or reserved?

```suggestion
    if (getCodeResponse.minted == true) {
      throw new CodeAlreadyMintedError(mintCode);
```

Reply:
I answer myself, no. Because is the response from Compass.